manager/volume.go

```diff
@@ -741,3 +741,39 @@ func (m *VolumeManager) UpdateDataLocality(name string, dataLocality types.DataL
 	logrus.Debugf("Updated volume %v data locality from %v to %v", v.Name, oldDataLocality, v.Spec.DataLocality)
 	return v, nil
 }
+
+func (m *VolumeManager) IncBackupFailedCount(name string) (v *longhorn.Volume, err error) {
```

Comment:
Remove these methods and add a metrics interface, then have metric class implementations in a datastore class.

example:
metrics.backup.succeeded(volumeName)
metrics.backup.failed(volumeName)

metrics is a singleton instance that contains references to the actual metric interface implementation.
this looks nice and makes it easy to split the metrics across different files.

Don't store these persistently, in the objects, this will create a lot of update and writes which are completely unnecessary. See below on how to store these in memory.

Also don't return any errors from these methods, metrics are best effort only.

Reply:
Follow your suggestion, I created a new collector for each type (volumeCollector, backupCollector, nodeCollector, etc..) and have a common baseCollector. 

> Don't store these persistently, in the objects, this will create a lot of update and writes which are completely unnecessary. See below on how to store these in memory.

We will discuss more about this on the meeting tommorow